File: backend/app/schemas/master_data.py
from __future__ import annotations

from pydantic import BaseModel, Field


class MasterOption(BaseModel):
    code: str
    label: str


class LinkedMasterOption(MasterOption):
    state_code: str | None = None
    district_code: str | None = None
    tehsil_code: str | None = None


class MasterListResponse(BaseModel):
    items: list[MasterOption] = Field(default_factory=list)


class LinkedMasterListResponse(BaseModel):
    items: list[LinkedMasterOption] = Field(default_factory=list)


CaseTypeMasterResponse = MasterListResponse
SpecialCaseMasterResponse = MasterListResponse
PartyNameSuffixResponse = MasterListResponse

File: frontend/src/types/masterData.ts
export interface MasterOption {
  code: string;
  label: string;
}

export interface LinkedMasterOption extends MasterOption {
  state_code?: string | null;
  district_code?: string | null;
  tehsil_code?: string | null;
}

export interface MasterListResponse {
  items: MasterOption[];
}

export interface LinkedMasterListResponse {
  items: LinkedMasterOption[];
}

export type CaseTypeMasterResponse = MasterListResponse;
export type SpecialCaseMasterResponse = MasterListResponse;
export type PartyNameSuffixResponse = MasterListResponse;
